client/src/components/MainContent.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react';
+import Websocket from 'react-websocket';
+import ComputerSysDataTable from './ComputerSysDataTable';
+import ComputerSysContainer from './ComputerSysContainer';
+import { WS_URL, fetchAllComputerSystems } from '../js/ApiHelper';
+import { Grid, Row, Col } from 'react-bootstrap';
+import { ClipLoader } from 'react-spinners';
+import { chartColors, getProvisionStateInfo } from '../js/ProvisionStateChartUtil';
+import rd3 from 'rd3';
+import FontAwesome from 'react-fontawesome';
+import 'font-awesome/css/font-awesome.css';
+import '../css/style.css';
+
+const PieChart = rd3.PieChart;
+
+/**
+ * Main component to display the application content.
+ */
+class MainContent extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			computerSystems: [],
+			isLoading: false,
+			isRowSelected: false,
+			row: null,
+			updateSelectedRow: false
+		};    
+	}
+
+	/**
+	 * Requests the server to grab the computer systems list.
+	 */
+	componentDidMount() {
+		this.setState({isLoading: true});
+
+		fetchAllComputerSystems(
+			(data) => this.setState({status: 'success', computerSystems: data[0].systems, isLoading: false}),
+			(error) => this.setState({status: 'error', isLoading: false}));
+	}
+	
+	/**
+	 * Handle the selection of a row in the ComputerSysDataTable component. By transfering this 
+	 * data we can then load the ComputerSysContainer component.
+	 */
+	onRowSelectCallback = (isRowSelected, row) => {
+		this.setState({
+			isRowSelected: isRowSelected,
+			row: row,
+			updateSelectedRow: false
+		});
+	}
+
+	handleData(data) {
+		const results = JSON.parse(data);
+		if (this.state.isRowSelected) {
+			this.setState({computerSystems: results[0].systems, updateSelectedRow: true});
+		} else {
+			this.setState({computerSystems: results[0].systems, updateSelectedRow: false});
+		}
+	}
+
+	render() {
+		const {status, computerSystems, isLoading, updateSelectedRow} = this.state;
+
+		if (isLoading) {
```

Comment:
Maybe the conditional returns pattern and can be wrapped up a bit with some `element variables`.

Reply:
True, in that way the return will be more clear and don't need to have multiple returns. Nice tip, thanks.